Sources/GroundState/py_variational_montecarlo.hpp

```diff
@@ -256,189 +319,91 @@ void AddVariationalMonteCarloModule(py::module &m) {
-
-                 data = vmc.compute_samples(sampler, 10000, 1000)
-                 ex, lv = vmc.expectation(data, psi, ham, return_locvals=True)
-                 var = vmc.variance(data, psi, ham, ex["Mean"], lv)
-               ```
-            )EOF");
-
-  using GradType1 = VectorXcd (*)(const vmc::Result &, AbstractMachine &,
-                                  const AbstractOperator &);
-  m_vmc.def("gradient", (GradType1)&vmc::Gradient, py::arg("vmc_data"),
-            py::arg("op"), py::arg("psi"), R"EOF(
-           Computes the gradient of the expecation value of a Hermitian operator
-           `op` with respect to the wavefunction parameters based on provided VMC
-           data.
+                  Args:
+                      sampler: sampler to use for Monte Carlo sweeps.
+                      n_samples: number of samples to record.
+                      n_discard: number of sweeps to discard.
+                      der_logs: Whether to calculate gradients of the logarithm
+                          of the wave function. `None` means don't compute,
+                          "normal" means compute, and "centered" means compute
+                          and then center.
 
-           Args:
-               vmc_data: The VMC result data.
-               psi: Machine represenation of the wavefunction.
-               op: Hermitian operator.
-            )EOF");
-
-  using GradType2 = VectorXcd (*)(const vmc::Result &, AbstractMachine &,
-                                  const AbstractOperator &, const VectorXcd &);
-  m_vmc.def("gradient", (GradType2)&vmc::Gradient, py::arg("vmc_data"),
-            py::arg("op"), py::arg("psi"), py::arg("locvals"), R"EOF(
-           Computes the gradient of the expecation value of a Hermitian operator
-           `op` with respect to the wavefunction parameters based on provided VMC
-           data.
-
-           Args:
-               vmc_data: The VMC result data.
-               psi: Machine represenation of the wavefunction.
-               op: Hermitian operator.
-               locvals: An array containing the local values of `op` for the
-                   given VMC data.
-            )EOF");
-
-  m_vmc.def("local_value", &vmc::LocalValue, py::arg("op"), py::arg("psi"),
-            py::arg("v"), R"EOF(
-           Computes the local value of the operator `op` in configuration `v`
-           which is defined as O_loc(v) = ⟨v|op|Ψ⟩ / ⟨v|Ψ⟩.
-
-           Args:
-               op: Hermitian operator.
-               psi: Machine represenation of the wavefunction.
-               v: Visible configuration.
-            )EOF");
+                  Returns:
+                      A `MCResult` object with all the data obtained during sampling.)EOF");
 
   m_vmc.def(
-      "local_values",
-      [](const vmc::Result &result, AbstractMachine &psi,
-         const AbstractOperator &op) {
-        return vmc::LocalValues(op, psi, result.SampleMatrix());
+      "gradient_of_expectation",
```

Comment:
I am still thinking how to structure these helper functions better... I mean, in python I'd expect just two functions:
```python 
s_results=compute_samples(...)
grad=gradient_of_expectation(my_hamiltonian, s_results...) 
```

I mean, intuitively I don't like the idea to specifically call `local_values` first, and basically enter into (unecessary) details about how the gradient of expectation is internally computed (i.e. first forming the local values)